### ISSUE TITLE
x/pkgsite: duplicate escaping on error page

### ISSUE THREAD

**Issue by findleyr** [golang/go#63968]
Via @rsc, who encountered the following broken escaping on pkgsite:
![image](https://github.com/golang/go/assets/57144380/0f3a74b9-71e4-4ba1-9e77-762691f3240a)

Repro steps:
1. Visit https://pkg.go.dev/rsc.io/tmp/nonexist
2. Click "Fetch" button.
3. Wait for the load to fail.

**Comment by findleyr** [comment]
Interestingly, the test appears to look for this escaping explicitly:
https://cs.opensource.google/go/x/pkgsite/+/master:internal/frontend/fetchserver/fetch_test.go;l=170;drc=ebe617b30c52774fccf9d36331cbff31e99aaf9e

**Comment by findleyr** [comment]
Hmm, I don't know if/how this ever worked. Perhaps @jba recalls?

[`fetch.ts`](https://cs.opensource.google/go/x/pkgsite/+/master:static/frontend/fetch/fetch.ts;l=16;drc=8e5e504098ab1a2d307a1300a817660a892604b1) uses `new DOMParser().parseFromString(...)`, but that doesn't unescape:
![image](https://github.com/golang/go/assets/57144380/7f72a419-6c75-46b7-afb4-4da8ca86a6be)

**Comment by hyangah** [comment]
![Screenshot 2024-02-20 at 1 30 40 PM](https://github.com/golang/go/assets/4999471/d464a2c1-3b2f-4d21-b46c-917cf4b7c041)
I guess this was fixed just by updating the pkgsite backend... last week.

**Comment by findleyr** [comment]
Looks like this is still happening, reported by @mvdan

**Comment by mvdan** [comment]
Yes; I went to https://pkg.go.dev/golang.org/x/tools/internal/ast/cursor, I was presented with the "request docs for this package", and when I clicked it, I eventually loaded a page which looked like the original here - a bunch of HTML starting with a `<div>`. I did not take a screenshot, and the page now returns a correct 404 not found page, but I briefly shared my screen at today's tools call and I think that got recorded :)

**Comment by findleyr** [comment]
I'm guessing that the async fetch operations polls an endpoint, and eventually escapes the error content before formatting on screen.
And this doesn't happen if the error is rendered in the original page.
